debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -231,11 +267,44 @@ protected SchemaChangeEvent getCreateTableEvent(RelationalSnapshotContext snapsh
     }
 
     @Override
-    protected Optional<String> getSnapshotSelect(SnapshotContext snapshotContext, TableId tableId) {
+    protected Optional<String> getSnapshotSelect(RelationalSnapshotContext snapshotContext, TableId tableId) {
+        String columnString = buildSelectColumns(connectorConfig.getConfig().getString(connectorConfig.COLUMN_BLACKLIST), snapshotContext.tables.forTable(tableId));
+
         long snapshotOffset = (Long) snapshotContext.offset.getOffset().get("scn");
         return Optional.of("SELECT * FROM " + quote(tableId) + " AS OF SCN " + snapshotOffset);
     }
 
+    /**
+     * This is to build "whitelisted" column list
+     * @param blackListColumnStr comma separated columns blacklist
+     * @param table the table
+     * @return column list for select
+     */
+    public static String buildSelectColumns(String blackListColumnStr, Table table) {
+        String columnsToSelect = "*";
+        if (blackListColumnStr != null && blackListColumnStr.trim().length() > 0
```

Comment:
Shouldn't use "blacklist", but see above.

Reply:
Code removed & re-implement as suggested in the other issue.